core/storage/trie/serialization/codec.hpp

```diff
@@ -50,6 +59,10 @@ namespace kagome::storage::trie {
      * @return hash of \param buf or \param buf if it is shorter than the hash
      */
     virtual common::Buffer merkleValue(const common::BufferView &buf) const = 0;
+    virtual outcome::result<common::Buffer> merkleValue(
+        const OpaqueTrieNode &node,
+        std::optional<StateVersion> version,
+        const ChildVisitor &child_visitor = NoopChildVisitor) const = 0;
```

Comment:
`TrieNode` caches its own hash or encoded bytes.
Users will read `node.cached_encoded` directly, because they know that this value is `Some`
```suggestion
```